### DISCUSSION TITLE
File version 4 is outside the supported range of your printer

### DISCUSSION THREAD

**Discussion by jcg0474** [sn4k3/UVtools#796] [General]
I suddenly started receiving this warning:

"Your printer and file format seems only to support the following versions: (2). However, the file comes with the version 4 set, which is outside the supported range.
Keeping that version may or not print on your machine, but can mislead UVtools to set and use features that your printer firmware doesn't support or vice versa leading to unexpected behaviors.

Do you want to change the version 4 to the latest supported version 2?
Yes: Change to version 2. (Highly recommended!)
No: Keep the version 4 while editing, but a latter full encode of the file will force the version 2."

Indeed, files no longer function properly on my Phrozen Sonic Mini. The Z-axis lowers to the bed, stops, and the machine either reboots itself or it raises the z-axis back up and requests that I try replugging the USB in.

I'm using Chitubox to generate the files, directly from Chitubox they work with no issues. It's only when I open them and then save them in UVTools that things go wonky. 

Any suggestions to fix this? I'm not sure what "version 2" and "version 4" means, I couldn't find anything about that in the Chitubox documentation.

**Comment by sn4k3** [comment]
Can you provide both files, before and after UVtools save?

**Comment by Hazoooka** [comment]
CTB version most likely.
I presume you have Phrozen Sonic Mini, the first release.
My Mini 8k supports CTB version 4, Mini 4K probably supports CTB version 3, your's is most likely version 2.
Correct me anyone if I'm wrong but this seems to be the case to me.

**Reply by jcg0474** [reply]
I tried an earlier version of CTB as well (1.8.1) with the same results.  I couldn't find a version further back than that.

**Reply by sn4k3** [reply]
CTB on UVtools can use from version 2 to 4 regardless the printer, so that make me think you are using the cbddlp format?

**Comment by Hazoooka** [comment]
> files no longer function properly on my Phrozen Sonic Mini

What was the slicer version the last time the files were properly interpreted?
Di you try to re-install the latest firmware for your printer?

**Comment by jcg0474** [comment]
Okay, further discoveries, this is going to be a lot.

The CTB files I'm providing are just an example model, what I actually want to "print" is a photolithography mask. To do so I'm using the pixel arithmetic tool in UVTools to make a pattern for all pixels in the layer. So far I've tested: CTB Version 1.8.1 files (A1), CTB Version 1.9.5 files (B1), CTB Version 1.8.1 files opened in UVTools and resaved (A2), CTB Version 1.9.5 files opened in UVTools and resaved (B2), CTB Version 1.8.1 files converted to photolithography mask (C1), CTB Version 1.9.5 files converted to photolithography mask (C2).

Of these, the ones that don't throw errors on the Phrozen Sonic mini are: A1, A2, B1, B2. So it appears that the Pixel arithmetic is doing something wonky?!?!? I've included all of the files thus far and some screenshots. At one point I had this working fine, then I updated the firmware, updated Chitubox, and now it just doesn't work. 

[CTB181.zip](https://github.com/sn4k3/UVtools/files/13591764/CTB181.zip)
[CTB195.zip](https://github.com/sn4k3/UVtools/files/13591769/CTB195.zip)
![Pixel arithmetic 1](https://github.com/sn4k3/UVtools/assets/44736366/cb276901-c9ef-4d16-9f32-fe3706109232)
![Pixel arithmetic 2](https://github.com/sn4k3/UVtools/assets/44736366/43e3ee72-149a-4115-890c-0bf412655c53)

**Reply by sn4k3** [reply]
Do not use patterns on those printers, it won't have memory to load the slice and will trigger error.
